### PR TITLE
update _changes API documentation for 2.0

### PR DESCRIPTION
**Pull request by willholley**
* use string-based sequence identifiers in examples
 * document the `pending` field in responses
 * add a new about partial ordering/idempotency

Fixes COUCHDB-3121

### CODE REVIEW

**Comment by kxepal** [comment]
LGFM

**Comment by rnewson** [comment]
`sequence identifiers` is a very late-minute terminology invention. We usually call these `update sequences` or `update sequence values`.

File: src/api/database/changes.rst [hunk 1]
```diff
@@ -95,7 +95,8 @@
                            - :mimetype:`text/plain; charset=utf-8`
     :>header ETag: Response hash is changes feed is `normal`
     :>header Transfer-Encoding: ``chunked``
-    :>json number last_seq: Last change sequence number
+    :>json string last_seq: Last change sequence identifier
```
Comment (rnewson) [inline src/api/database/changes.rst]: it can be any valid JSON, classifying this item as a string is the same mistake as classifying it as number.

**Comment by willholley** [comment]
@rnewson sure - I couldn't see an example other than `sequence number` in the CouchDB docs. Cloudant uses `sequence identifiers` in the documentation so I went with that as a first pass. I'll change it to `update sequences` or similar.

File: src/json-structure.rst [hunk 2]
```diff
@@ -72,11 +72,13 @@ Changes information for a database
 +--------------------------------+---------------------------------------------+
 | Field                          | Description                                 |
 +================================+=============================================+
-| last_seq                       | Last change sequence number                 |
+| last_seq                       | Last change sequence identifier             |
```
Comment (rnewson) [inline src/json-structure.rst]: "last update sequence"

File: src/json-structure.rst [hunk 3]
```diff
@@ -72,11 +72,13 @@ Changes information for a database
...
++--------------------------------+---------------------------------------------+
+| pending                        | Count of remaining items in the feed        |
 +--------------------------------+---------------------------------------------+
 | results [array]                | Changes made to a database                  |
 +--------------------------------+---------------------------------------------+
-|         seq                    | Update sequence number                      |
+|         seq                    | Update sequence identifier                  |
```
Comment (rnewson) [inline src/json-structure.rst]: "Update sequence"

File: src/api/database/changes.rst [hunk 4]
```diff
@@ -104,7 +105,7 @@
 
     :json array changes: List of document`s leafs with single field ``rev``
     :json string id: Document ID
-    :json number seq: Update sequence number
+    :json string seq: Update sequence
```
Comment (rnewson) [inline src/api/database/changes.rst]: json

**Comment by willholley** [comment]
thanks all for the feedback. I've pushed a new update which addresses the comments and adds a new section on the `_selector` filter.

**Comment by rnewson** [comment]
+1 after that last comment is addressed

**Pull request merged**